### PR TITLE
Added kubernetes resource exists function 

### LINKED ISSUE #725: Implement k8s_resource_exists function
k8s_list and k8s_get were recently added, we should have a k8s_resource_exists func too.

### PR DESCRIPTION
**Pull request by Sanskarzz**
Added kubernetes resource exists function
Fixes #725

### CODE REVIEW

**Comment by eddycharly** [comment]
It’s the same as exists ?

**Comment by Sanskarzz** [comment]
Sorry my mistake

**Comment by Sanskarzz** [comment]
I shared an approach on slack check it

File: pkg/runner/check/functions/k8s.go [hunk 1]
```diff
@@ -6,9 +6,47 @@ import (
...
+	err = client.List(context.TODO(), &unstructured.UnstructuredList{
+		Object: obj.Object,
+	}, ctrlclient.InNamespace(""))
+
+	if err == nil {
+		return true, nil
+	}
+	if apierrors.IsNotFound(err) {
+		return false, nil
+	}
```
Comment (eddycharly) [inline pkg/runner/check/functions/k8s.go:46]: @Sanskarzz we don't need to get the list, the mapping should be enough to determine if the resource exists or not :)

Comment (Sanskarzz) [reply pkg/runner/check/functions/k8s.go]: okay

**Review by eddycharly — APPROVED** [review]
LGTM

**Pull request merged**